hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieFileIndex.scala

```diff
@@ -342,6 +342,16 @@ object HoodieFileIndex extends Logging {
     if (listingModeOverride != null) {
       properties.setProperty(DataSourceReadOptions.FILE_INDEX_LISTING_MODE_OVERRIDE.key, listingModeOverride)
     }
+    val partitionColumns = metaClient.getTableConfig.getPartitionFields
+    if (partitionColumns.isPresent) {
+      // NOTE: Multiple partition fields could have non-encoded slashes in the partition value.
+      //       We might not be able to properly parse partition-values from the listed partition-paths.
```

Comment:
Hey @codope, if `URL_ENCODE_PARTITIONING` is enabled, it's safe to use lazy file listing?